russh-keys/src/key.rs

```diff
@@ -282,6 +301,31 @@ impl PublicKey {
                     .verify(buffer, &signature)
                     .is_ok()
             }
+            PublicKey::P521(ref public) => {
+                const FIELD_LEN: usize =
+                    <p521::NistP521 as p256::elliptic_curve::Curve>::FieldBytesSize::USIZE;
+                let mut reader = sig.reader(0);
+                let mut read_field = || -> Option<p521::FieldBytes> {
+                    let f = reader.read_mpint().ok()?;
+                    let f = f.strip_prefix(&[0]).unwrap_or(f);
+                    let mut result = [0; FIELD_LEN];
+                    if f.len() > FIELD_LEN {
+                        return None;
+                    }
+                    #[allow(clippy::indexing_slicing)] // length is known
+                    result[FIELD_LEN - f.len()..].copy_from_slice(f);
+                    // Some(result.into())
+                    Some(GenericArray::clone_from_slice(&result)) // ew
```

Comment:
GenericArray strangely does not implement `From<[u8; 66]>`. Any ideas?

Reply:
Yeah, `p521` is using an older version of `generic_arary` where `From<[T; N]>` is only implemented for N up to 64